test/integration-tests/Uniswap.test.ts

```diff
@@ -1196,6 +1206,480 @@ describe('Uniswap V2 and V3 Tests:', () => {
+
+      let usdcSpent = bobUSDCBalanceBefore.sub(bobUSDCBalanceAfter)
+      let wethSpent = bobWETHBalanceBefore.sub(bobWETHBalanceAfter)
+
+      // check that the USDC and WETH were spent
+      expect(usdcSpent).to.be.gte(0)
+      expect(wethSpent).to.be.gte(0)
+
+      const receipt = await tx.wait()
+
+      const transferEvent = receipt.events?.find((event) => event.event === 'IncreaseLiquidity')
+
+      tokenId = transferEvent?.args?.tokenId
+    })
+
+    describe('erc721permit', () => {
+      it('erc721 permit succeeds', async () => {
+        const { v, r, s } = await getPermitNFTSignature(bob, v3NFTPositionManager, router.address, tokenId, MAX_UINT)
+
+        planner.addCommand(CommandType.ERC721_PERMIT, [ADDRESS_THIS, tokenId, MAX_UINT, v, r, s])
+
+        // bob permits the router to spend token
+        await executeRouter(planner)
+
+        expect((await v3NFTPositionManager.positions(tokenId)).operator).to.eq(router.address)
+      })
+
+      it('erc721 permit also succeeds when passing in actual address', async () => {
+        const { v, r, s } = await getPermitNFTSignature(bob, v3NFTPositionManager, router.address, tokenId, MAX_UINT)
+
+        planner.addCommand(CommandType.ERC721_PERMIT, [router.address, tokenId, MAX_UINT, v, r, s])
+
+        // bob permits the router to spend token
+        await executeRouter(planner)
+
+        expect((await v3NFTPositionManager.positions(tokenId)).operator).to.eq(router.address)
+      })
+
+      it('only owner of the token can permit another address', async () => {
+        const { v, r, s } = await getPermitNFTSignature(eve, v3NFTPositionManager, router.address, tokenId, MAX_UINT)
+        planner.addCommand(CommandType.ERC721_PERMIT, [ADDRESS_THIS, tokenId, MAX_UINT, v, r, s])
+
+        // bob is trying to permit the router using eve's signature
+        await expect(executeRouter(planner)).to.be.revertedWith('Unauthorized')
+      })
+    })
+
+    describe('decrease liquidity', () => {
+      it('decrease liquidity succeeds', async () => {
+        // first we need to permit the router to spend the nft
+        const { v, r, s } = await getPermitNFTSignature(bob, v3NFTPositionManager, router.address, tokenId, MAX_UINT)
+        planner.addCommand(CommandType.ERC721_PERMIT, [ADDRESS_THIS, tokenId, MAX_UINT, v, r, s])
+
+        let position = await v3NFTPositionManager.positions(tokenId)
+        let liquidity = position.liquidity
+
+        const decreaseParams = {
+          tokenId: tokenId,
+          liquidity: liquidity,
+          amount0Min: 0,
+          amount1Min: 0,
+          deadline: MAX_UINT,
+        }
+
+        const encodedDecreaseCall = encodeDecreaseLiquidity(decreaseParams)
+
+        planner.addCommand(CommandType.V3_POSM_CALL, [encodedDecreaseCall])
+        await executeRouter(planner)
+
+        position = await v3NFTPositionManager.positions(tokenId)
+        liquidity = position.liquidity
+
+        expect(liquidity).to.eq(0)
```

Comment:
would be great to check that `tokensOwed` in the position manager gas increased too to show that there are tokens to claim after the decrease